scripts/globals/job_utils/puppetmaster.lua

```diff
@@ -0,0 +1,410 @@
+{
+    [xi.item.CAN_OF_AUTOMATON_OIL   ] = 1, -- Automaton Oil
+    [xi.item.CAN_OF_AUTOMATON_OIL_P1] = 2, -- Automaton Oil + 1
+    [xi.item.CAN_OF_AUTOMATON_OIL_P2] = 3, -- Automaton Oil + 2
+    [xi.item.CAN_OF_AUTOMATON_OIL_P3] = 4  -- Automaton Oil + 3
+}
+
+-- On Ability Check Overdrive
+xi.job_utils.puppetmaster.onAbilityCheckOverdrive = function(player, target, ability)
+    local pet = player:getPet()
+    if not pet then
+        return xi.msg.basic.REQUIRES_A_PET, 0
+    elseif not pet:isAutomaton() then
+        return xi.msg.basic.NO_EFFECT_ON_PET, 0
+    else
+        ability:setRecast(math.max(0, ability:getRecast() - player:getMod(xi.mod.ONE_HOUR_RECAST) * 60))
+        return 0, 0
+    end
+end
+
+-- On Ability Use Overdrive
+xi.job_utils.puppetmaster.onAbilityUseOverdrive = function(player, target, ability)
+    player:addStatusEffect(xi.effect.OVERDRIVE, 0, 0, 60)
+
+    return xi.effect.OVERDRIVE
+end
+
+-- On Ability Check Activate
+xi.job_utils.puppetmaster.onAbilityCheckActivate = function(player, target, ability)
+    if player:getPet() ~= nil then
+        return xi.msg.basic.ALREADY_HAS_A_PET, 0
+    elseif not player:canUseMisc(xi.zoneMisc.PET) then
+        return xi.msg.basic.CANT_BE_USED_IN_AREA, 0
+    elseif player:isExceedingElementalCapacity() then
+        return xi.msg.basic.AUTO_EXCEEDS_CAPACITY, 0
+    end
+
+    return 0, 0
+end
+
+-- On Ability Use Activate
+xi.job_utils.puppetmaster.onAbilityUseActivate = function(player, target, ability)
+    xi.pet.spawnPet(player, xi.petId.AUTOMATON)
+
+    local pet = player:getPet()
+
+    if pet then
+        local jpValue = player:getJobPointLevel(xi.jp.AUTOMATON_HP_MP_BONUS)
+        pet:addMod(xi.mod.HP, jpValue * 10)
+        pet:addMod(xi.mod.MP, jpValue * 5)
+    end
+end
+
+-- On Ability Check Deux Ex Automata
+xi.job_utils.puppetmaster.onAbilityCheckDeuxExAutomata = function(player, target, ability)
+    if player:getPet() ~= nil then
+        return xi.msg.basic.ALREADY_HAS_A_PET, 0
+    elseif not player:canUseMisc(xi.zoneMisc.PET) then
+        return xi.msg.basic.CANT_BE_USED_IN_AREA, 0
+    else
+        local jpValue = player:getJobPointLevel(xi.jp.DEUS_EX_AUTOMATA_RECAST)
+
+        ability:setRecast(ability:getRecast() - jpValue)
+
+        return 0, 0
+    end
+end
+
+-- On Ability Use Deux Ex Automata
+xi.job_utils.puppetmaster.onAbilityUseDeuxExAutomata = function(player, target, ability)
+    xi.pet.spawnPet(player, xi.petId.AUTOMATON)
+    local pet = player:getPet()
+
+    if pet then
+        local percent = math.floor((player:getMainLvl() / 3)) / 100
+        pet:setHP(math.max(pet:getHP() * percent, 1))
+        pet:setMP(pet:getMP() * percent)
+    end
+end
+
+--Repair Function - TODO: verify that this function should be here and not on the items "Enhances Repair"
+-- https://www.bg-wiki.com/ffxi/Repair
+local function removeStatus(pet)
+    for _, effectId in ipairs(removableEffectIds) do
+        if pet:delStatusEffect(effectId) then
```

Comment:
are you just moving code around?
Could rewrite this as `return pet:delStatusEffect(effectId)` (nit), but don't need to if the intent is just to move the code

Reply:
At this point, I am making no distinct changes to any of the code as far as calculations or process. However, there were a few places where I was able to trim some of the fat due to repeating checks or as an example, repair and maintenance. They have the same table of removable effects.